ts/features/wallet/privative/component/PrivativeWalletPreview.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { Option } from "fp-ts/lib/Option";
+import { Image, ImageStyle, StyleProp } from "react-native";
+import { Body } from "../../../../components/core/typography/Body";
+import { navigateToPrivativeDetailScreen } from "../../../../store/actions/navigation";
+import { GlobalState } from "../../../../store/reducers/types";
+import { PrivativePaymentMethod } from "../../../../types/pagopa";
+import { IOStyles } from "../../../../components/core/variables/IOStyles";
+import { useImageResize } from "../../onboarding/bancomat/screens/hooks/useImageResize";
+import { CardLayoutPreview } from "../../component/CardLayoutPreview";
+import unknownGdo from "../../../../../img/wallet/unknown-gdo.png";
+
+type OwnProps = {
+  privative: PrivativePaymentMethod;
+};
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps> &
+  OwnProps;
+
+const BASE_IMG_W = 160;
+const BASE_IMG_H = 20;
+
+const renderRight = (props: Props, size: Option<[number, number]>) =>
+  size.fold(
+    <Image
+      source={unknownGdo}
+      style={{ width: 40, height: 40, resizeMode: "contain" }}
+      key={"unknownGdoLogo"}
+      testID={"unknownGdoLogo"}
+    />,
+    imgDim => {
+      const imageUrl = props.privative?.cardLogo;
+
+      const imageStyle: StyleProp<ImageStyle> = {
+        width: imgDim[0],
+        height: imgDim[1],
+        resizeMode: "contain"
+      };
+      return imageUrl ? (
+        <Image
+          source={{ uri: imageUrl }}
+          style={imageStyle}
+          key={"gdoLogo"}
+          testID={"gdoLogo"}
+        />
+      ) : (
+        <Image
```

Comment:
To avoid code duplication could it be this image a const and refer to it when it is needed?

```typescript
const renderRight = (props: Props, size: Option<[number, number]>) => {
  const fallbackImage = (
    <Image
      source={unknownGdo}
      style={{ width: 40, height: 40, resizeMode: "contain" }}
      key={"unknownGdoLogo"}
      testID={"unknownGdoLogo"}
    />
  );
  return size.fold(fallbackImage, imgDim => {
    const imageUrl = props.privative?.cardLogo;

    const imageStyle: StyleProp<ImageStyle> = {
      width: imgDim[0],
      height: imgDim[1],
      resizeMode: "contain"
    };
    return imageUrl ? (
      <Image
        source={{ uri: imageUrl }}
        style={imageStyle}
        key={"gdoLogo"}
        testID={"gdoLogo"}
      />
    ) : (
      fallbackImage
    );
  });
};
```

Reply:
Sure, modified in [7402c36](https://github.com/pagopa/io-app/pull/2878/commits/7402c36b88f9e3a4a952bb2ca778ff0cb5e1b399)